### ISSUE TITLE
Quick list of install issues on Amazon linux

### ISSUE THREAD

**Issue by Alex-Ikanow** [cantino/huginn#126]
This was starting from version 12.08 of Amazon Linux (which is pre-installed with ruby-1.8: which I deleted, and then installed "ruby19", "ruby19gem" and "ruby19-devel" from the Amazon repository, which installs ruby19-1.9.3.484-31.55.amzn1.x86_64 etc):

(Referencing these instructions: https://gist.github.com/mjhea0/b6b58eefc38985380ff9)

Before Step 2 I had to install the following RPMs via yum:
* yum install gcc-c++
* yum install curl-devel
* yum install mysql-devel

Step 4: my ruby19 RPM installs "/usr/bin/bundler" not "/usr/bin/bundle" so I ran "bundler" instead of "bundle" everywhere

Step 8, when running the "rake" command, I ran into a number of problems from the "bundler install" steps:
* It complained about rake version, I edited Gemfile.lock and changed the Rake to "1.10.1" (from "1.10.0")
* It complained about "mysql2/mysql2" not being found. I had to do: "cp /usr/local/share/gems1.9/gems/mysql2-0.3.14/ext/mysql2/mysql2.so  /usr/local/share/gems1.9/gems/mysql2-0.3.14/lib/mysql2"
* It complained about there being no JS runtime environment, I added the following line to Gemfile:
** gem 'therubyracer', :platforms => :ruby
* It complained about "nokogiri/nokogiri" not being found, I did "cp /usr/local/share/gems1.9/gems/nokogiri-1.6.0/ext/nokogiri/nokogiri.so /usr/local/share/gems1.9/gems/nokogiri-1.6.0/lib/nokogiri/"
* It complained about "Unable to load the EventMachine C extension; To use the pure-ruby reactor, require 'em/pure_ruby'"
** cp /usr/local/share/gems1.9/gems/eventmachine-1.0.3/ext/rubyeventmachine.so /usr/local/share/gems1.9/gems/eventmachine-1.0.3/lib/
* "cannot load such file -- ruby_http_parser"
** cp /usr/local/share/gems1.9/gems/http_parser.rb-0.5.3/ext/ruby_http_parser/ruby_http_parser.so /usr/local/share/gems1.9/gems/http_parser.rb-0.5.3/lib/
* "cannot load such file -- minitest/unit"
** Added "gem 'minitest'" to Gemfile

Step 10: when I ran "rake db:migrate", I got  "cannot load such file -- bcrypt_ext"
* cp /usr/local/share/gems1.9/gems/bcrypt-ruby-3.1.1/ext/mri/bcrypt_ext.so /usr/local/share/gems1.9/gems/bcrypt-ruby-3.1.1/lib/

In summary there were 3 things that went wrong (in lots of different ways!)
* Some gems weren't installed correctly, the ".so" files didn't get put in the right place
* Some incorrect version numbers
* Some gems were missing (eg therubyracer, minitest)

Finally when I ran it, the JSON didn't parse until I fixed multi_json to 1.7.8 (in Gemfile and Gemfile.lock) and reran "bundler install" and also then ran "bundler upgrade jsonpath" (I don't know if both those steps were necessary).

**Comment by cantino** [comment]
We have official install docs now!
